neevaco/neeva-android
# Sync cached login cookies back into the browser cookie manager

If we have a cached session cookie (which can only be the login one at the moment, because it's stored in SharedPrefs), sync it down to the browser at initialization before proceeding with the rest of the CookieManager initialization.  See the comment in `SessionToken.kt` for details.

Also:
* Update the tests to account for the changed behavior.
* Update `CoroutineScopeRule` so that it'll crash and print out the stack when a coroutine crashes.

* Using `singleInstance` instead of `singleTask` doesn't provide a whole lot of benefit, and actually causes each of our instrumentation tests to pointlessly hang by 45 seconds.  Switch back to using `singleTask` (things still work as expected).  (https://github.com/android/android-test/issues/1115)

## Checklists

### How was this tested?

### Manual test cases
1. Built a version of the app before https://github.com/neevaco/neeva-android/pull/1001 landed
2. Log into that version of the app
3. Build a version of the app off of main + this PR
4. Confirm that the user is still logged in

## Issues addressed
* Fixes https://github.com/neevaco/neeva-android/issues/1043
